testing/test_faulthandler.py

```diff
@@ -135,3 +136,25 @@ def test():
         result.stdout.no_fnmatch_line(warning_line)
     result.stdout.fnmatch_lines("*1 passed*")
     assert result.ret == 0
+
+
+def test_get_stderr_fileno_invalid_fd(monkeypatch) -> None:
+    """Test for faulthandler being able to handle invalid file descriptors for stderr (#8249)."""
+    from _pytest.faulthandler import FaultHandlerHooks
+
+    class StdErrWrapper(io.StringIO):
+        """
+        Mimic ``twisted.logger.LoggingFile`` to simulate returning an invalid file descriptor.
+
+        https://github.com/twisted/twisted/blob/twisted-20.3.0/src/twisted/logger/_io.py#L132-L139
+        """
+
+        def fileno(self):
+            return -1
+
+    wrapper = StdErrWrapper()
+    monkeypatch.setattr("sys.stderr", wrapper)
```

Comment:
Since this test itself runs under pytest, it's probably safer to scope the stderr changes, like so:


```py
with pytest.MonkeyPatch.context() as mp:
   mp.setattr("sys.stderr", wrapper)
    # test here...
```

This way by the time the test is finished the stderr is restored.

Reply:
Thanks again. I've implemented it like that and just amended the patch.